constraint/pkg/apis/templates/v1/constrainttemplate_types.go

```diff
@@ -49,7 +49,7 @@ type Validation struct {
 	// +kubebuilder:pruning:PreserveUnknownFields
 	OpenAPIV3Schema *apiextensionsv1.JSONSchemaProps `json:"openAPIV3Schema,omitempty"`
 	// +kubebuilder:default=false
-	LegacySchema bool `json:"legacySchema,omitempty"`
+	LegacySchema *bool `json:"legacySchema,omitempty"` // *bool allows for "unset" state
```

Comment:
nit: add "which we need to apply appropriate defaults"

Reply:
done